2 - ListsLogicAndIterations/Code/LogicAndInterations.cs

```diff
@@ -1,36 +1,96 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace ListsLogicAndIterations.Code
 {
 	public static class LogicAndInterations
 	{
 		public static int For_SumTwoListsAndAllValueThatAreNotFour(int[] numberArray, List<int> numberList)
 		{
-			throw new NotImplementedException();
+            int result = 0;
+            foreach(int i in numberArray)
+            {
+                if (i != 4)
+                    result += i;
+            }
+            foreach(int i in numberList)
+            {
+                if (i != 4)
+                    result += i;
+            }
+            return result;
+            
+            //throw new NotImplementedException();
 		}
 
 		public static int ForEach_SumTwoListsAndAllValueThatAreNotFour(List<int> numberList)
 		{
-			throw new NotImplementedException();
+            int result = 0;
+
+            foreach (int i in numberList)
+            {
+                if (i != 4)
+                result += i;
+            }
+            return result;
+
+            //throw new NotImplementedException();
 		}
 
 		public static int DoWhile_SumValuesWhileCurrentValueIsNot4_ThenStop(List<int> numberList)
 		{
-			throw new NotImplementedException();
+            int result = 0;
+            int i = 0;
+            do
+            {
+                result += numberList[i];
+                if (numberList[i] == 4)
+                    i = numberList.Count;
+                else i++;
+            }
+            while (i < numberList.Count);
+
+            return result;
+
+
+           // throw new NotImplementedException();
 		}
 
 		public static int While_SumValuesWhileCurrentValueIsNot4_ThenStop(List<int> numberList)
 		{
-			throw new NotImplementedException();
+            int i = 0;
+            int result = 0;
+            while (i < numberList.Count)
+            {
+                result += numberList[i];
+                if (numberList[i] != 4)
+                    i++;
+                else
+                    i = numberList.Count;
```

Comment:
I would not suggest doing `i = numberList.Count;` - if you want to exit the loop you can use `break;`